en/debug/jtag_pins.md

```diff
@@ -0,0 +1,130 @@
+# JTAG Hardware Connections for PX4 Hardware (Debug Port)
+
+PX4 runs on hardware that exposes a [JTAG interface in SW mode](https://en.wikipedia.org/wiki/JTAG), which can be used for on-hardware debugging.
+
+This topic explains how to connect to the JTAG interface on various boards.
+Using JTAG for debugging is then covered in other topics (e.g [MCU Eclipse/J-Link Debugging for PX4](../debug/eclipse_jlink.md)).
+
+> **Tip** The Segger [J-Link EDU Mini](https://www.segger.com/products/debug-probes/j-link/models/j-link-edu-mini/) is an inexpensive J-Link adapter.
+  [![Segger JLink EDU Mini Pinout](../../assets/debug/jlink_edu_mini.jpg)](ttps://www.segger.com/products/debug-probes/j-link/models/j-link-edu-mini/)
+
+
+## JTAG SW Modes Pins
+
+PX4 requires the following pins for JTAG.
+The pins should be connected on the corresponding pins of the flight controller board and JTAG device.
+
+Pin | Signal Type | Description
+--- | --- | ---
+VTref | Input | Target reference voltage. Used to check if the target has power, to create the logic-level reference for the input comparators and to control the output logic levels to the target. It is normally fed from Vdd of the target board and must not have a series resistor.
+SWDIO | I/O | Single bi-directional data pin. A pull-up resistor is required. ARM recommends 100 kOhms.
+SWCLK | Output | Clock signal to target CPU. It is recommended that this pin is pulled to a defined state on the target board. Typically connected to TCK of the target CPU.
+GND | - | Ground pin. Should be connected to GND in the target system.
+
+PX4 also supports these pins, if available (e.g. on FMUv5x).
+
+Pin | Signal Type | Description
+--- | --- | ---
+SWO | Input | Serial Wire Output trace port (Optional, not required for SWD communication.)
+nRESET | I/O | Target CPU reset signal. Typically connected to the RESET pin of the target CPU, which is typically called `nRST`.
+TRACECLK | Input | Input trace clock. Trace clock = 1/2 CPU clock.
+TRACEDATA[0] | Input | Input Trace data pin 0 (called TRACED0 in Pixhawk FMUv5x)
+
+>  **Note** The information above is reproduced from [J-Link / J-Trace User Guide: UM08001 > 18.1.2 Pinout for SW Mode](https://www.segger.com/downloads/jlink/UM08001) (Software Version: 6.54).
+
+
+## Pixhawk Series
+
+[Pixhawk series](https://docs.px4.io/master/en/flight_controller/pixhawk_series.html) boards provide access to JTAG through the **Pixhawk Debug Port**.
+The physical connector and pin order for this port varies across Pixhawk FMU versions.
+
+> **Note** The **Pixhawk Debug Port** also exposes a UART for access to the [System Console](../debug/system_console.md) (a console that provides low-level access to the system, debug output and analysis of the system boot process).
+
+<!--
```

Comment:
@davids5 
1. Do FMUv2 and 3 have debug ports?
2. If so, then what connector do they use (Hirose DF13), and what is the layout?

Reply:
Yes. Use ARM 10 pin JTAG connector and a Special cable to get the console.  DF13 to  FTDI 3 wires. 
![image](https://user-images.githubusercontent.com/1945821/68936533-a200e580-074f-11ea-9295-2eff8ba611cb.png)

Reply:
@davids5 Thanks. Re the ARM 10pin, that gets mentioned in [docs here](https://docs.px4.io/master/en/flight_controller/pixhawk.html#parts--housings) but where is the actual connector on the board? Do you have to take off the cover?

Reply:
@hamishwillee Yes. The case has to be removed to access the connector for the FMU and IO.
In most cases you have to solder on the connectors

FMU is in center (highlighted yellow)
IO is next to the PWM

![image](https://user-images.githubusercontent.com/1945821/69134656-b2c19c00-0a6c-11ea-80a1-e19ae30762a2.png)

Reply:
@davids5 How are these numbered? My assumption is that the little square indicates 1, and they are numbered 1-5, 6-10, with each pin being ordered as shown on the left.

![image](https://user-images.githubusercontent.com/5368500/69198266-3fad4800-0b88-11ea-9b0d-d4270c622467.png)

Reply:
Googleing ARM 10 pin. https://www.google.com/search?client=ms-android-tmus-us-revc&sxsrf=ACYBGNRS_6Ne8zkJtNCMk_LBCL5yBHUWDw%3A1574212891193&ei=G5XUXdeZC5L--gSVxIigAw&q=arm+10+pin+pin+out&oq=arm+10+pin+pin+out&gs_l=mobile-gws-wiz-serp.3..0i13i30j0i8i13i30l2j33i160.4262.7876..8527...0.1..0.258.1329.0j6j2......0....1.........0i71j0j0i67j0i22i30j33i22i29i30j33i299.YW9MxSQ7bFE#imgdii=StG8YrFZIHFmmM:&imgrc=YqJDfEDkf8Uf_M: